autoextract_poet/providers.py

```diff
@@ -0,0 +1,49 @@
+from typing import Type
+
+from scrapy.http import Request
+from scrapy.statscollectors import StatsCollector
+from scrapy_poet.page_input_providers import (
+    PageObjectInputProvider,
+    provides,
+)
+
+from autoextract_poet.inputs import (
+    AutoExtractProductResponseData,
+    AutoExtractProductListResponseData,
+)
+from autoextract_poet.utils import request
+
+
+class AutoExtractProvider(PageObjectInputProvider):
+
+    page_type: str
+    page_input_class: Type
+
+    def __init__(self, request: Request, stats: StatsCollector):
+        self.request = request
+        self.stats = stats
+
+    async def __call__(self):
+        try:
+            # FIXME: how to configure if you want full HTML or not?
```

Comment:
We could receive Scrapy Settings on the `__init__` method, but do you think that's really necessary for now?

Reply:
I'm using this in the spyder where I'm always requiring HTML, but the general case will be not requesting HTML (by default it is not supplied for regular accounts). I see three ways of configuring that:
* By configuring the provider from settings
* By configuring the provider from request meta
* By deducting it by annotating something as `AEResponseData`

Probably the best is the last one, but it requires changes in the providers infrastructure to support it. And there are complexities like what happen if you request `AEProductPage`, `AEArticlePage` and `AEResponseData`. What is inside `AEResponseData` in this case, the HTML from the product or from the article extraction?

Reply:
How common is it to request the HTML response from the AutoExtract API? Thinking about skipping it here on this pull request and address it later...

Reply:
Another approach could be by using types.
We could've a `AutoExtractFullHtmlProvider`  and `ProductResponseFullHtmlData`.
Not sure if it is better, but it gives control on the request level to enable/disable HTML response.
For example, for product listing I don't want the HTML only what AE returns, but for product pages I want everything.
The problem is that it creates more classes and granularity, which can be a bit convoluted